network/v1/types.go

```diff
@@ -54,33 +79,54 @@ type ClusterNetworkList struct {
+// HostSubnet
+// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$`
+type HostSubnetEgressIP string
+
+// HostSubnetEgressCIDR represents one egress CIDR from which to assign IP addresses for this node
+// represented by the HostSubnet
+// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])/([0-9]|[12][0-9]|3[0-2])$`
+type HostSubnetEgressCIDR string
+
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // HostSubnet describes the container subnet network on a node. The HostSubnet object must have the
 // same name as the Node object it corresponds to.
+// +kubebuilder:printcolumn:name="Host",type=string,JSONPath=`.host`,description="The name of the node"
+// +kubebuilder:printcolumn:name="Host IP",type=string,JSONPath=`.hostIP`,description="The IP address to be used as a VTEP by other nodes in the overlay network"
+// +kubebuilder:printcolumn:name="Subnet",type=string,JSONPath=`.subnet`,description="The CIDR range of the overlay network assigned to the node for its pods"
+// +kubebuilder:printcolumn:name="Egress CIDRs",type=string,JSONPath=`.egressCIDRs`,description="The network egress CIDRs"
+// +kubebuilder:printcolumn:name="Egress IPs",type=string,JSONPath=`.egressIPs`,description="The network egress IP addresses"
 type HostSubnet struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// Host is the name of the node. (This is the same as the object's name, but both fields must be set.)
+	// +kubebuilder:validation:Pattern=`^[a-z0-9.-]+$`
 	Host string `json:"host" protobuf:"bytes,2,opt,name=host"`
+
 	// HostIP is the IP address to be used as a VTEP by other nodes in the overlay network
+	// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$`
 	HostIP string `json:"hostIP" protobuf:"bytes,3,opt,name=hostIP"`
+
 	// Subnet is the CIDR range of the overlay network assigned to the node for its pods
+	// +kubebuilder:validation:Pattern=`^(([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[0-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])/([0-9]|[12][0-9]|3[0-2])$`
 	Subnet string `json:"subnet" protobuf:"bytes,4,opt,name=subnet"`
 
 	// EgressIPs is the list of automatic egress IP addresses currently hosted by this node.
 	// If EgressCIDRs is empty, this can be set by hand; if EgressCIDRs is set then the
 	// master will overwrite the value here with its own allocation of egress IPs.
 	// +optional
-	EgressIPs []string `json:"egressIPs,omitempty" protobuf:"bytes,5,rep,name=egressIPs"`
+	EgressIPs []HostSubnetEgressIP `json:"egressIPs,omitempty" protobuf:"bytes,5,rep,name=egressIPs"`
```

Comment:
I'm not quite sure this will be backwards compatible, @deads2k or @sttts do you know?

Reply:
and even if it's JSON-API-compatible, it's not golang-API-compatible and so it would be simpler to just not change it

Reply:
The downside is that we can't add a validation on each string in the slice.

Reply:
When we pull in the changes to openshift/api, won't it be sufficient to make the corresponding golang change in the SDN and ovn repos?

Reply:
Yes, although then there's more divergence between 4.4 and earlier. But I didn't realize that there was no way to validate the individual strings otherwise

Reply:
@sttts can you comment on @soltysh's question above?

Reply:
As an API on the wire it is compatible. Go might have an hickup. But that's ok.